common/models/model-definition.js

```diff
@@ -75,18 +75,18 @@ function ready(ModelDefinition) {
     assert(relation.embed && relation.embed.as, 'embed requires "as"');
     switch (relation.embed.as) {
       case 'object':
-        assert(relation.embed.key || relation.embed.keyFn,
-          'embed as object requires "key" or "keyFn"');
+        assert(relation.embed.key || relation.embed.keySetter,
+          'embed as object requires "key" or "keySetter"');
         result = {};
         relatedData.forEach(function(related) {
           var Definition = app.models[relation.model];
           var key;
           if (relation.embed.key) {
             key = related[relation.embed.key];
           }
-          var keyFn = relation.embed.keyFn;
-          if (keyFn && typeof Definition[keyFn] === 'function') {
-            key = Definition[keyFn](related);
+          var keySetter = relation.embed.keySetter;
```

Comment:
I think this should be `keyGetter`, because we want to build a key from the data stored in our domain entity, shouldn't it?